<p>
  Unused parameters are misleading. Whatever the value passed to such parameters is, the behavior will be the same.
</p>

<h2>Noncompliant Code Example</h2>
<pre>
function doSomething($a, $b) { // "$a" is unused
  return compute($b);
}
</pre>

<h2>Compliant Solution</h2>
<pre>
function doSomething($b) {
  return compute($b);
}
</pre>

<h2>Exceptions</h2>
<p>Methods overriding or implementing another one should be excluded</p>
<pre>
/*
 * @inheritdoc
 */
function doSomething($a, $b) {     // no issue reported on $b
  compute($a);
}
</pre>
