<p>
  Not using curly braces could be error-prone in some cases.
</p>

<h2>Noncompliant Code Example</h2>
<pre>
// the two statements seems to be attached to the if statement whereas this is the case only for the first one:
if (condition)
  executeSomething();
  checkSomething();
</pre>

<h2>Compliant Solution</h2>
<pre>
if (condition) {
  executeSomething();
  checkSomething();
}
</pre>
