<p>
  The <code>eval</code> function is a way to run arbitrary code at run-time.
</p>
<p>According to the PHP documentation</p>
<blockquote>The eval() language construct is very dangerous because it allows execution of arbitrary PHP code. Its use thus is discouraged. If you have carefully verified that there is no other option than to use this construct, pay special attention not to pass any user provided data into it without properly validating it beforehand.</blockquote>

<h2>Noncompliant Code Example</h2>
<pre>
eval($code_to_be_dynamically_executed)
</pre>
